src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -686,22 +685,16 @@ void RTPSParticipantImpl::setup_output_traffic()
         num_send_buffers += m_receiverResourcelist.size();
     }
 
-    if (num_network_buffers == 0)
+    if (network_buffers_config.initial == 0 && network_buffers_config.increment == 0)
     {
-        // 16 network buffers per send buffer by default
-        num_network_buffers = 16;
-    }
-
-    if (inc_network_buffers == 0)
-    {
-        // New buffers dynamically allocated when needed during RTPS message creation
-        // 16 buffers increment by default
-        inc_network_buffers = 16;
+        // 16 network buffers per send buffer preallocated by default
+        network_buffers_config.initial = 16;
+        // New buffers dynamically allocated when needed during RTPS message creation by default
+        network_buffers_config.increment = 16;
     }
 
     // Create buffer pool
-    send_buffers_.reset(new SendBuffersManager(num_send_buffers, allow_growing_buffers, num_network_buffers,
-            inc_network_buffers));
+    send_buffers_.reset(new SendBuffersManager(num_send_buffers, allow_growing_buffers, network_buffers_config));
```

Comment:
And use the config on `m_att` directly.

```suggestion
    send_buffers_.reset(new SendBuffersManager(num_send_buffers, allow_growing_buffers, m_att.allocation.send_buffers.network_buffers_config));
```